webapp/channels/src/components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -81,17 +82,25 @@ function makeGetPluginSchema() {
                         component,
                         showTitle: customComponents[key] ? customComponents[key].options.showTitle : false,
                     };
-                });
+                }) as Array<Partial<AdminDefinitionSetting & PluginSetting>>;
```

Comment:
Why do we need the as?

Reply:
This was generating some problems to make everything consistent with the `PluginSetting` type. it is not ideal, but the PR is very complex and I didn't want to keep digging on it.